### DISCUSSION TITLE
Cannot install package ip6tables-extra. Redmi ax6000 openwrt 23.05.5

### DISCUSSION THREAD

**Discussion by tatburlak1** [bol-van/zapret#965] [Q&A]
Всё делал по инструкции но почему-то возникла такая ошибка, sdk вроде у них есть для моего роутера, но что-то не понимаю как ей пользоваться, я правильно понял что через неё надо делать?
Collected errors:
 * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-reject
 * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-reject found, but incompatible with the architectures configured                                     * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-ipt
 * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-ipt found, but incompatible with the architectures configured                                        * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-log                                              * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-log found, but incompatible with the architectures configured                                        * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-core                                            * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-core found, but incompatible with the architectures configured                                      * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nfnetlink                                           * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nfnetlink found, but incompatible with the architectures configured                                     * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-ipset                                           * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-ipset found, but incompatible with the architectures configured                                     * satisfy_dependencies_for: Cannot satisfy the following dependencies for ipset:                                *      kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec)                                                   * opkg_install_cmd: Cannot install package ipset.       * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-reject6                                          * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-reject6 found, but incompatible with the architectures configured                                    * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-conntrack                                        * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-conntrack found, but incompatible with the architectures configured                                  * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-conntrack6                                       * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-conntrack6 found, but incompatible with the architectures configured                                 * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-nat                                              * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-nat found, but incompatible with the architectures configured                                        * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-log6                                             * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-log6 found, but incompatible with the architectures configured                                       * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-crypto-hash                                         * pkg_hash_fetch_best_installation_candidate: Packages for kmod-crypto-hash found, but incompatible with the architectures configured                                   * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-crypto-crc32c                                       * pkg_hash_fetch_best_installation_candidate: Packages for kmod-crypto-crc32c found, but incompatible with the architectures configured                                 * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-lib-crc32c                                          * pkg_hash_fetch_best_installation_candidate: Packages for kmod-lib-crc32c found, but incompatible with the architectures configured                                    * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nft-core                                            * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nft-core found, but incompatible with the architectures configured                                      * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nft-compat                                          * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nft-compat found, but incompatible with the architectures configured                                    * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-extra                                           * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-extra found, but incompatible with the architectures configured                                     * satisfy_dependencies_for: Cannot satisfy the following dependencies for iptables-mod-extra:                   *      kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec)                                                   * opkg_install_cmd: Cannot install package iptables-mod-extra.                                                  * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nfnetlink-queue                                     * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nfnetlink-queue found, but incompatible with the architectures configured                               * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-nfqueue                                         * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-nfqueue found, but incompatible with the architectures configured                                   * satisfy_dependencies_for: Cannot satisfy the following dependencies for iptables-mod-nfqueue:                 *      kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec)                                                   * opkg_install_cmd: Cannot install package iptables-mod-nfqueue.                                                * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-lib-textsearch                                      * pkg_hash_fetch_best_installation_candidate: Packages for kmod-lib-textsearch found, but incompatible with the architectures configured                                * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-conntrack                                       * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-conntrack found, but incompatible with the architectures configured                                 * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-filter                                          * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-filter found, but incompatible with the architectures configured                                    * satisfy_dependencies_for: Cannot satisfy the following dependencies for iptables-mod-filter:                  *      kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec)                                                   * opkg_install_cmd: Cannot install package iptables-mod-filter.                                                 * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-ipopt                                           * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-ipopt found, but incompatible with the architectures configured                                     * satisfy_dependencies_for: Cannot satisfy the following dependencies for iptables-mod-ipopt:                   *      kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec)                                                   * opkg_install_cmd: Cannot install package iptables-mod-ipopt.                                                  * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-conncount                                        * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-conncount found, but incompatible with the architectures configured                                  * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-conntrack-extra                                 * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-conntrack-extra found, but incompatible with the architectures configured                           * satisfy_dependencies_for: Cannot satisfy the following dependencies for iptables-mod-conntrack-extra:         *      kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec)                                                   * opkg_install_cmd: Cannot install package iptables-mod-conntrack-extra.                                        * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-nat6                                             * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-nat6 found, but incompatible with the architectures configured                                       * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-nat                                             * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-nat found, but incompatible with the architectures configured                                       * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-nf-ipt6                                             * pkg_hash_fetch_best_installation_candidate: Packages for kmod-nf-ipt6 found, but incompatible with the architectures configured                                       * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ip6tables                                           * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ip6tables found, but incompatible with the architectures configured                                     * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ipt-nat6                                            * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ipt-nat6 found, but incompatible with the architectures configured                                      * satisfy_dependencies_for: Cannot satisfy the following dependencies for ip6tables-mod-nat:                    *      kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec)                                                   * opkg_install_cmd: Cannot install package ip6tables-mod-nat.                                                   * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec) for kmod-ip6tables-extra                                     * pkg_hash_fetch_best_installation_candidate: Packages for kmod-ip6tables-extra found, but incompatible with the architectures configured                               * satisfy_dependencies_for: Cannot satisfy the following dependencies for ip6tables-extra:                      *      kernel (= 5.15.167-1-03ba5b5fee47f2232a088e3cd9832aec)                                                   * opkg_install_cmd: Cannot install package ip6tables-extra.                                                    could not install prerequisites

**Comment by tatburlak1** [comment]
Ссылка на SDK https://downloads.openwrt.org/releases/23.05.5/targets/mediatek/filogic/openwrt-sdk-23.05.5-mediatek-filogic_gcc-12.3.0_musl.Linux-x86_64.tar.xz

**Comment by bol-van** [comment]
Это китаец.
Есть ушлые китайцы, которые берут железки на похожем чипсете, но там немного железо отличается.
Допиливают на коленке openwrt, предустанавливают базовый софт и заливают, потом продают.
Рабочего репозитория кернел модулей нет.
Идет расчет на идиота, который только в веб морду тычет

Нужно брать модели только с официальной поддержкой openwrt. Желательно без предустановленной openwrt, потому что это вероятно обман, описанный выше

**Comment by bol-van** [comment]
Сегодня столкнулся с аналогичным девайсом.
Была китайская запоротая прошивка, не загружалась, но был sysupgrade образ от китайцев.
Модель можно посмотреть на рабочей системе в файле /etc/board.json.
Если рабочей системы нет, можно распаковать ядро из sysupgrade. Это проще всего сделать FARом на винде. Ctrl+pagedown входит как в архив.
Дальше вытащить оттуда dtb. Есть питон скрипт на гитхабе. И декомпилировать его device tree compiler (dtc).
Оттуда легко прочитать имя модели. Оказалось 	Livinet ZR-3020.
Официального образа на openwrt не нашлось, но на китайском форке immortalwrt нашлось. Встало без проблем

**Comment by tatburlak1** [comment]
Так он был без openwrt, это уже я его прошивал.

**Comment by tatburlak1** [comment]
https://downloads.openwrt.org/releases/23.05.5/targets/mediatek/filogic/openwrt-23.05.5-mediatek-filogic-xiaomi_redmi-router-ax6000-stock-initramfs-factory.ubi сама прошивка, прошилася без проблем, как раз этот роутер поддерживается. Потом через Люси загрузил уже эту https://downloads.openwrt.org/releases/23.05.5/targets/mediatek/filogic/openwrt-23.05.5-mediatek-filogic-xiaomi_redmi-router-ax6000-stock-squashfs-sysupgrade.bin

**Comment by tatburlak1** [comment]
Помогите разобраться как собрать или собрать через SDK который у них указан, возможно просто не сталкивались с такой версией?

**Comment by bol-van** [comment]
На этом SDK

https://mirror-03.infra.openwrt.org/releases/23.05.5/targets/mediatek/filogic/openwrt-sdk-23.05.5-mediatek-filogic_gcc-12.3.0_musl.Linux-x86_64.tar.xz

без проблем собралось по этой инструкции

docs/compile/build_howto_openwrt.txt

**Comment by tatburlak1** [comment]
Спасибо, Попробую

**Comment by bol-van** [comment]
Я плохо прочитал первый пост. Думал вам надо собрать проги из zapret под свою платформу

iptables для новых openwrt не нужен. там используются nftables
iptables установить можно из репозитория. если они не ставятся - значит сборка не официальный релиз скорее всего
но смысла не вижу. если только из-за какой-то инструкции в инете, то не стоит

**Comment by tatburlak1** [comment]
Да я это делаю чисто из инструкции в интернете, поэтому выбрал первый вариант iptables. Попробую выбрать nftables

**Comment by tatburlak1** [comment]
Выбрал nftables и получил ошибку:
Cannot install package kmod-nft-queue.                                                     could not install prerequisites.

Значит попробовать собрать через SDK? Или не в этом проблема?

**Comment by bol-van** [comment]
Не может быть такого, если образ нормальный.
Что выдают команды ?
```
cat /etc/os-release
cat /etc/opkg/distfeeds.conf
cat /proc/version
opkg info kernel

```

**Comment by tatburlak1** [comment]
`root@Dom4a:/tmp/zapret-v69.8# cat /etc/os-release       NAME="OpenWrt"                                          VERSION="23.05.5"                                       ID="openwrt"                                            ID_LIKE="lede openwrt"                                  PRETTY_NAME="OpenWrt 23.05.5"                           VERSION_ID="23.05.5"                                    HOME_URL="https://openwrt.org/"                         BUG_URL="https://bugs.openwrt.org/"                     SUPPORT_URL="https://forum.openwrt.org/"                BUILD_ID="r24106-10cc5fcd00"                            OPENWRT_BOARD="mediatek/filogic"                        OPENWRT_ARCH="aarch64_cortex-a53"                       OPENWRT_TAINTS=""                                       OPENWRT_DEVICE_MANUFACTURER="OpenWrt"                   OPENWRT_DEVICE_MANUFACTURER_URL="https://openwrt.org/"  OPENWRT_DEVICE_PRODUCT="Generic"                        OPENWRT_DEVICE_REVISION="v0"                            OPENWRT_RELEASE="OpenWrt 23.05.5 r24106-10cc5fcd00"     root@Dom4a:/tmp/zapret-v69.8# cat /etc/opkg/distfeeds.conf                                                      src/gz openwrt_core https://downloads.openwrt.org/releases/23.05.5/targets/mediatek/filogic/packages            src/gz openwrt_base https://downloads.openwrt.org/releases/23.05.5/packages/aarch64_cortex-a53/base             src/gz openwrt_luci https://downloads.openwrt.org/releases/23.05.5/packages/aarch64_cortex-a53/luci             src/gz openwrt_packages https://downloads.openwrt.org/releases/23.05.5/packages/aarch64_cortex-a53/packages     src/gz openwrt_routing https://downloads.openwrt.org/releases/23.05.5/packages/aarch64_cortex-a53/routing       src/gz openwrt_telephony https://downloads.openwrt.org/releases/23.05.5/packages/aarch64_cortex-a53/telephony   
root@Dom4a:/tmp/zapret-v69.8# cat /proc/version         Linux version 5.15.167 (builder@buildhost) (aarch64-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r24106-10cc5fcd00) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 SMP Mon Sep 23 12:34:46 2024                                         
root@Dom4a:/tmp/zapret-v69.8# opkg info kernel          
root@Dom4a:/tmp/zapret-v69.8#`

**Comment by bol-van** [comment]
Текст надо форматировать, так очень сложно читать.
Но главное здесь - нет вывода opkg info kernel
Что-то испорчено или левое установлено, и в этом разбираться смысла нет.

Надо взять официальный sysupgrade image и сделать sysupgrade -n sysupgrade_filename
